### PR TITLE
Fixes #11282 - Search on numeric facts is broken if using "~"

### PR DESCRIPTION
**Pull request by imriz**

### CODE REVIEW

**Comment by dLobatog** [comment]
@imriz I just tried searching for "uptime_days ~ 4" on 1.7, 1.8, 1.9, couldn't reproduce on any. I tried on both the Hosts list and the Fact Values list (/facts), is there anywhere else I should try? Otherwise the patch looks fine to me.

cc @amirfefer as you reported this

**Comment by imriz** [comment]
@eLobato I don't think my original patch was merged to the stable branches, so it shouldn't be broken there either.

**Comment by ohadlevy** [comment]
@eLobato indeed - its a followup to 3f8e6c33

**Comment by dLobatog** [comment]
@ohadlevy @imriz You mean this is only broken on develop? I think I tested that too, mind to provide some data to verify it?

File: test/unit/host_test.rb [hunk 1]
```diff
@@ -1367,6 +1367,10 @@ def teardown
       assert_equal hosts.count, 1
       assert_equal ["num001.example.com"], hosts.map { |h| h.name }.sort
 
+      hosts = Host::Managed.search_for("facts.memory_mb ~ 64498")
```
Comment (dLobatog) [inline test/unit/host_test.rb:1370]: This test is passing on the current develop branch, without the change above in `scoped_search_extensions`. Could you add some test that tests the functionality added here?

Comment (imriz) [inline test/unit/host_test.rb:1370]: You tried with postgres? It failed on my side before the fix

Comment (dLobatog) [inline test/unit/host_test.rb:1370]: Good catch, it does indeed fail _only_ on pg.

**Comment by dLobatog** [comment]
Merged as e7ee509d04f9f839e3a95830f29de6af9948c782, thanks @imriz!

**Pull request closed without merging**